### ISSUE TITLE
Always keep pinned tabs on screen

### ISSUE THREAD

**Issue by marc-groundctl** [qutebrowser/qutebrowser#4419]
It would be nice to make pinned tabs behave more like Firefox's pinned tabs (even if only as an option); always onscreen and in the corner.  I use pinned tabs for sites which indicate status with their favicon, and being always able to see that would be nice.

**Comment by The-Compiler** [comment]
I don't think this is possible without completely reimplementing our own tab bar, which is not something I'd like to do. @jgkamat what's your take on this?

**Comment by jgkamat** [comment]
Florian Bruhin writes:

> I don't think this is possible without completely reimplementing our own tab bar, which is not something I'd like to do. @jgkamat what's your take on this?

I agree with this. It also would be more interesting to implement, because
pinned tabs can be anywhere in the tab bar, not just on the left (so we can't
just scroll the right tabs).

> It would be nice to make pinned tabs behave more like Firefox's pinned tabs (even if only as an option); always onscreen and in the corner. I use pinned tabs for sites which indicate status with their favicon, and being always able to see that would be nice.

It probably would be possible to add a setting to disable tab scrolling
entirely, ie: instead of having a min tab size, keep shrinking tabs past their
usability point chrome-style. Would that work for you?

**Comment by marc-groundctl** [comment]
That would be better, but seeing tab title is useful.  Would it be possible to have pinned tabs in their own tab bar on the side and the rest of the tabs in the main bar?

**Comment by jgkamat** [comment]
> That would be better, but seeing tab title is useful. Would it be possible to have pinned tabs in their own tab bar on the side and the rest of the tabs in the main bar?

This sounds like something that would be better done as a plugin (if the
relevant APIs are available), as I don't think many people would use it, and
it could cause a lot of bugs and edge cases.
